readthedocs/doc_builder/environments.py

```diff
@@ -432,6 +372,44 @@ def _escape_command(self, cmd):
             command = command.replace(f"\\${variable}", f"${variable}")
         return command
 
+    def dump_and_get_final_command(self):
+        """
+        Dump the command into a shell script and return the correct command to execute.
+
+        Create a shell script file called `/readthedocs.sh` with the content of
+        the command to be executed by the Docker container.
+        This file includes an extra binary PATH in case we are running
+        Python/Conda/Mamba/etc binaries.
+
+        We dump the command into a particular shell script to allow running multi-line commands
+        in ``build.jobs`` and ``build.commands``.
+        """
+        bin_path = f"{self.bin_path}:" if self.bin_path else ""
+        command = " ".join(
+            self._escape_command(part) if self.escape_command else part
+            for part in self.command
+        )
+        script_content = f"""
+#!/bin/bash
+
+export PATH={bin_path}$PATH
+{command}
+"""
+        tarstream = io.BytesIO()
+        tar = tarfile.TarFile(fileobj=tarstream, mode="w")
+        script_data = script_content.encode("utf8")
+        tarinfo = tarfile.TarInfo(name="readthedocs.sh")
```

Comment:
I think we probably want to use a random name for this file, so we don't end up with some kind of weird docker error running old commands?